<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Section 8 Appendix 1: Primer on INLA | High-resolution spatial modelling of population welfare in El Salvador, a coding tutorial</title>
  <meta name="description" content="This is report provides a tutorial to conduct proof-of-concept high-resolution maps of poverty, average income and literacy. The emphasis is put on providing replicable code exemples.">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Section 8 Appendix 1: Primer on INLA | High-resolution spatial modelling of population welfare in El Salvador, a coding tutorial />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is report provides a tutorial to conduct proof-of-concept high-resolution maps of poverty, average income and literacy. The emphasis is put on providing replicable code exemples." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Section 8 Appendix 1: Primer on INLA | High-resolution spatial modelling of population welfare in El Salvador, a coding tutorial />
  
  <meta name="twitter:description" content="This is report provides a tutorial to conduct proof-of-concept high-resolution maps of poverty, average income and literacy. The emphasis is put on providing replicable code exemples." />
  

<meta name="author" content="Xavier Vollenweider, Claudio Bosco">


<meta name="date" content="2020-05-29">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="discussion.html">
<link rel="next" href="data.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<script src="libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/Proj4Leaflet-1.0.1/proj4-compressed.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.0.2/leaflet.js"></script>
<script src="libs/leaflet-providers-1.1.17/leaflet-providers.js"></script>
<script src="libs/leaflet-providers-plugin-2.0.2/leaflet-providers-plugin.js"></script>
<script src="libs/plotly-binding-4.8.0/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-1.39.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-1.39.2/plotly-latest.min.js"></script>
<link href="libs/lfx-pulse-icon-0.1.0/lfx-pulse-icon-prod.css" rel="stylesheet" />
<script src="libs/lfx-pulse-icon-0.1.0/lfx-pulse-icon-prod.js"></script>
<script src="libs/lfx-pulse-icon-0.1.0/lfx-pulse-icon-bindings.js"></script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Table of Content</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="overview.html"><a href="overview.html"><i class="fa fa-check"></i><b>2</b> Overview</a><ul>
<li class="chapter" data-level="2.1" data-path="overview.html"><a href="overview.html#first-look-at-the-data"><i class="fa fa-check"></i><b>2.1</b> First look at the data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="explo.html"><a href="explo.html"><i class="fa fa-check"></i><b>3</b> Data exploration and preselection of covariates</a><ul>
<li class="chapter" data-level="3.1" data-path="explo.html"><a href="explo.html#outcome-variabes-distribution-and-outliers"><i class="fa fa-check"></i><b>3.1</b> Outcome variabes: distribution and outliers</a><ul>
<li class="chapter" data-level="3.1.1" data-path="explo.html"><a href="explo.html#income"><i class="fa fa-check"></i><b>3.1.1</b> Income</a></li>
<li class="chapter" data-level="3.1.2" data-path="explo.html"><a href="explo.html#literacy"><i class="fa fa-check"></i><b>3.1.2</b> Literacy</a></li>
<li class="chapter" data-level="3.1.3" data-path="explo.html"><a href="explo.html#moderate-poverty"><i class="fa fa-check"></i><b>3.1.3</b> Moderate Poverty</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="explo.html"><a href="explo.html#unsupervised-dimension-reduction-based-on-distance-correlation"><i class="fa fa-check"></i><b>3.2</b> Unsupervised dimension reduction based on distance correlation</a><ul>
<li class="chapter" data-level="3.2.1" data-path="explo.html"><a href="explo.html#obtaining-a-dataframe-with-standardized-covariates"><i class="fa fa-check"></i><b>3.2.1</b> Obtaining a dataframe with standardized covariates</a></li>
<li class="chapter" data-level="3.2.2" data-path="explo.html"><a href="explo.html#dimension-reduction-based-on-distance-correlation"><i class="fa fa-check"></i><b>3.2.2</b> Dimension reduction based on distance correlation</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="explo.html"><a href="explo.html#correlations-between-predictors-and-the-outcome-variables"><i class="fa fa-check"></i><b>3.3</b> Correlations between predictors and the outcome variables</a><ul>
<li class="chapter" data-level="3.3.1" data-path="explo.html"><a href="explo.html#correlation-with-income"><i class="fa fa-check"></i><b>3.3.1</b> Correlation with income</a></li>
<li class="chapter" data-level="3.3.2" data-path="explo.html"><a href="explo.html#correlation-with-literacy"><i class="fa fa-check"></i><b>3.3.2</b> Correlation with literacy</a></li>
<li class="chapter" data-level="3.3.3" data-path="explo.html"><a href="explo.html#correlation-with-poverty"><i class="fa fa-check"></i><b>3.3.3</b> Correlation with poverty</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="simple-model.html"><a href="simple-model.html"><i class="fa fa-check"></i><b>4</b> Fitting a simple model</a><ul>
<li class="chapter" data-level="4.1" data-path="simple-model.html"><a href="simple-model.html#loading-the-data"><i class="fa fa-check"></i><b>4.1</b> Loading the data</a></li>
<li class="chapter" data-level="4.2" data-path="simple-model.html"><a href="simple-model.html#area-data"><i class="fa fa-check"></i><b>4.2</b> Area data</a><ul>
<li class="chapter" data-level="4.2.1" data-path="simple-model.html"><a href="simple-model.html#spatial-dependency-with-area-data-data"><i class="fa fa-check"></i><b>4.2.1</b> Spatial dependency with <em>area data</em> data</a></li>
<li class="chapter" data-level="4.2.2" data-path="simple-model.html"><a href="simple-model.html#store-the-data-for-modelling"><i class="fa fa-check"></i><b>4.2.2</b> Store the data for modelling</a></li>
<li class="chapter" data-level="4.2.3" data-path="simple-model.html"><a href="simple-model.html#split-the-sample-into-a-8020-training-and-validation-sets"><i class="fa fa-check"></i><b>4.2.3</b> Split the sample into a 80/20 training and validation sets</a></li>
<li class="chapter" data-level="4.2.4" data-path="simple-model.html"><a href="simple-model.html#formula-specifying-the-relationship-between-the-outcome-variable-and-the-fixed-and-spatial-random-effects"><i class="fa fa-check"></i><b>4.2.4</b> Formula specifying the relationship between the outcome variable and the fixed and spatial random effects</a></li>
<li class="chapter" data-level="4.2.5" data-path="simple-model.html"><a href="simple-model.html#fit-the-model-on-the-training-set"><i class="fa fa-check"></i><b>4.2.5</b> Fit the model on the training set</a></li>
<li class="chapter" data-level="4.2.6" data-path="simple-model.html"><a href="simple-model.html#validate-the-model-on-the-validation-set"><i class="fa fa-check"></i><b>4.2.6</b> Validate the model on the validation set</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="simple-model.html"><a href="simple-model.html#point-referenced-data"><i class="fa fa-check"></i><b>4.3</b> Point-referenced data</a><ul>
<li class="chapter" data-level="4.3.1" data-path="simple-model.html"><a href="simple-model.html#spatial-dependency-with-point-referenced-data-data"><i class="fa fa-check"></i><b>4.3.1</b> Spatial dependency with <em>point-referenced data</em> data</a></li>
<li class="chapter" data-level="4.3.2" data-path="simple-model.html"><a href="simple-model.html#store-the-data-for-modelling-in-a-stack"><i class="fa fa-check"></i><b>4.3.2</b> Store the data for modelling in a stack</a></li>
<li class="chapter" data-level="4.3.3" data-path="simple-model.html"><a href="simple-model.html#formula-specifying-the-relationship-between-the-outcome-variable-and-the-fixed-and-spatial-random-effects-1"><i class="fa fa-check"></i><b>4.3.3</b> Formula specifying the relationship between the outcome variable and the fixed and spatial random effects</a></li>
<li class="chapter" data-level="4.3.4" data-path="simple-model.html"><a href="simple-model.html#fit-the-model-on-the-training-set-1"><i class="fa fa-check"></i><b>4.3.4</b> Fit the model on the training set</a></li>
<li class="chapter" data-level="4.3.5" data-path="simple-model.html"><a href="simple-model.html#validate-the-model-on-the-validation-set-1"><i class="fa fa-check"></i><b>4.3.5</b> Validate the model on the validation set</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="simple-model.html"><a href="simple-model.html#k-fold-cross-validation"><i class="fa fa-check"></i><b>4.4</b> <em>K-fold</em> cross-validation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="stepwise.html"><a href="stepwise.html"><i class="fa fa-check"></i><b>5</b> Stepwise Covariates Selection</a><ul>
<li class="chapter" data-level="5.1" data-path="stepwise.html"><a href="stepwise.html#step-1-split-the-sample-into-a-602020-training-validation-and-test-sets"><i class="fa fa-check"></i><b>5.1</b> Step 1: Split the sample into a 60/20/20 training, validation and test sets</a></li>
<li class="chapter" data-level="5.2" data-path="stepwise.html"><a href="stepwise.html#bundle-step-2-to-4-in-a-function"><i class="fa fa-check"></i><b>5.2</b> Bundle step 2 to 4 in a function</a></li>
<li class="chapter" data-level="5.3" data-path="stepwise.html"><a href="stepwise.html#bundle-step-5-to-7-in-a-loop-ran-in-parallel"><i class="fa fa-check"></i><b>5.3</b> Bundle step 5 to 7 in a loop ran in parallel</a></li>
<li class="chapter" data-level="5.4" data-path="stepwise.html"><a href="stepwise.html#step-8-inspect-the-results-and-select-specification"><i class="fa fa-check"></i><b>5.4</b> Step 8: Inspect the results and select specification</a></li>
<li class="chapter" data-level="5.5" data-path="stepwise.html"><a href="stepwise.html#step-9-final-test"><i class="fa fa-check"></i><b>5.5</b> Step 9 Final test</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="summary-of-resuls.html"><a href="summary-of-resuls.html"><i class="fa fa-check"></i><b>6</b> Results summary</a><ul>
<li class="chapter" data-level="6.1" data-path="summary-of-resuls.html"><a href="summary-of-resuls.html#goodness-of-fit-at-the-segmento-level"><i class="fa fa-check"></i><b>6.1</b> Goodness of fit at the <em>segmento</em> level</a></li>
<li class="chapter" data-level="6.2" data-path="summary-of-resuls.html"><a href="summary-of-resuls.html#segmento-level-map"><i class="fa fa-check"></i><b>6.2</b> <em>Segmento</em> level map</a></li>
<li class="chapter" data-level="6.3" data-path="summary-of-resuls.html"><a href="summary-of-resuls.html#goodness-of-fit-at-the-municipio-level"><i class="fa fa-check"></i><b>6.3</b> Goodness of fit at the <em>Municipio</em> level</a><ul>
<li class="chapter" data-level="6.3.1" data-path="summary-of-resuls.html"><a href="summary-of-resuls.html#obtion-1-aggregating-results-at-the-municipio-level-using-all-segmentos"><i class="fa fa-check"></i><b>6.3.1</b> Obtion 1: Aggregating results at the municipio level using all <em>segmentos</em></a></li>
<li class="chapter" data-level="6.3.2" data-path="summary-of-resuls.html"><a href="summary-of-resuls.html#option-2-aggregating-results-at-the-municipio-level-using-only-segmentos-in-representative-municipios"><i class="fa fa-check"></i><b>6.3.2</b> Option 2: Aggregating results at the municipio level using only <em>segmentos</em> in representative municipios</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="discussion.html"><a href="discussion.html"><i class="fa fa-check"></i><b>7</b> Discussion</a></li>
<li class="chapter" data-level="8" data-path="theory.html"><a href="theory.html"><i class="fa fa-check"></i><b>8</b> Appendix 1: Primer on INLA</a><ul>
<li class="chapter" data-level="8.1" data-path="theory.html"><a href="theory.html#area-data-vs-point-referenced-data"><i class="fa fa-check"></i><b>8.1</b> Area data vs point-referenced data</a><ul>
<li class="chapter" data-level="8.1.1" data-path="theory.html"><a href="theory.html#bayesian-approach"><i class="fa fa-check"></i><b>8.1.1</b> Bayesian approach</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="theory.html"><a href="theory.html#model-formulation"><i class="fa fa-check"></i><b>8.2</b> Model formulation</a><ul>
<li class="chapter" data-level="8.2.1" data-path="theory.html"><a href="theory.html#a-simple-model-not-accounting-for-spatial-dependence"><i class="fa fa-check"></i><b>8.2.1</b> A simple model not accounting for spatial dependence</a></li>
<li class="chapter" data-level="8.2.2" data-path="theory.html"><a href="theory.html#a-spatial-model"><i class="fa fa-check"></i><b>8.2.2</b> A spatial model</a></li>
<li class="chapter" data-level="8.2.3" data-path="theory.html"><a href="theory.html#estimating-the-model-with-aeral-data"><i class="fa fa-check"></i><b>8.2.3</b> Estimating the model with aeral data</a></li>
<li class="chapter" data-level="8.2.4" data-path="theory.html"><a href="theory.html#estimating-the-model-with-point-referenced-data"><i class="fa fa-check"></i><b>8.2.4</b> Estimating the model with point-referenced data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>9</b> Appendix 2: Data Pre-processing</a><ul>
<li class="chapter" data-level="9.1" data-path="data.html"><a href="data.html#sources"><i class="fa fa-check"></i><b>9.1</b> Data sources</a></li>
<li class="chapter" data-level="9.2" data-path="data.html"><a href="data.html#loading-the-survey"><i class="fa fa-check"></i><b>9.2</b> Loading the survey</a></li>
<li class="chapter" data-level="9.3" data-path="data.html"><a href="data.html#loading-the-vectors"><i class="fa fa-check"></i><b>9.3</b> Loading the vectors</a><ul>
<li class="chapter" data-level="9.3.1" data-path="data.html"><a href="data.html#loading-the-administrative-boundaries"><i class="fa fa-check"></i><b>9.3.1</b> Loading the administrative boundaries</a></li>
<li class="chapter" data-level="9.3.2" data-path="data.html"><a href="data.html#loading-the-shapefiles-containing-the-fields-used-as-covariates"><i class="fa fa-check"></i><b>9.3.2</b> Loading the shapefiles containing the fields used as covariates</a></li>
<li class="chapter" data-level="9.3.3" data-path="data.html"><a href="data.html#loading-point-referenced-data-from-tables"><i class="fa fa-check"></i><b>9.3.3</b> Loading point-referenced data from tables</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="data.html"><a href="data.html#loading-the-rasters"><i class="fa fa-check"></i><b>9.4</b> Loading the rasters</a><ul>
<li class="chapter" data-level="9.4.1" data-path="data.html"><a href="data.html#lights-at-night"><i class="fa fa-check"></i><b>9.4.1</b> Lights at night</a></li>
<li class="chapter" data-level="9.4.2" data-path="data.html"><a href="data.html#precipitation"><i class="fa fa-check"></i><b>9.4.2</b> Precipitation</a></li>
<li class="chapter" data-level="9.4.3" data-path="data.html"><a href="data.html#soil-degradation"><i class="fa fa-check"></i><b>9.4.3</b> Soil degradation</a></li>
<li class="chapter" data-level="9.4.4" data-path="data.html"><a href="data.html#vegetation-greenness"><i class="fa fa-check"></i><b>9.4.4</b> Vegetation greenness</a></li>
<li class="chapter" data-level="9.4.5" data-path="data.html"><a href="data.html#temperature"><i class="fa fa-check"></i><b>9.4.5</b> Temperature</a></li>
<li class="chapter" data-level="9.4.6" data-path="data.html"><a href="data.html#slope"><i class="fa fa-check"></i><b>9.4.6</b> Slope</a></li>
<li class="chapter" data-level="9.4.7" data-path="data.html"><a href="data.html#other-rasters"><i class="fa fa-check"></i><b>9.4.7</b> Other rasters</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="data.html"><a href="data.html#rasters-calculation"><i class="fa fa-check"></i><b>9.5</b> Rasters calculation</a><ul>
<li class="chapter" data-level="9.5.1" data-path="data.html"><a href="data.html#lights-at-night-1"><i class="fa fa-check"></i><b>9.5.1</b> Lights at night</a></li>
<li class="chapter" data-level="9.5.2" data-path="data.html"><a href="data.html#precipitation-1"><i class="fa fa-check"></i><b>9.5.2</b> Precipitation</a></li>
<li class="chapter" data-level="9.5.3" data-path="data.html"><a href="data.html#vegetation-greenness-1"><i class="fa fa-check"></i><b>9.5.3</b> Vegetation greenness</a></li>
<li class="chapter" data-level="9.5.4" data-path="data.html"><a href="data.html#soil-degradation-1"><i class="fa fa-check"></i><b>9.5.4</b> Soil degradation</a></li>
<li class="chapter" data-level="9.5.5" data-path="data.html"><a href="data.html#temperature-1"><i class="fa fa-check"></i><b>9.5.5</b> Temperature</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="data.html"><a href="data.html#distance-calculation"><i class="fa fa-check"></i><b>9.6</b> Distance calculation</a><ul>
<li class="chapter" data-level="9.6.1" data-path="data.html"><a href="data.html#distance-to-public-services"><i class="fa fa-check"></i><b>9.6.1</b> Distance to public services</a></li>
<li class="chapter" data-level="9.6.2" data-path="data.html"><a href="data.html#distance-to-businesses"><i class="fa fa-check"></i><b>9.6.2</b> Distance to businesses</a></li>
<li class="chapter" data-level="9.6.3" data-path="data.html"><a href="data.html#distance-to-natural-features"><i class="fa fa-check"></i><b>9.6.3</b> Distance to natural features</a></li>
<li class="chapter" data-level="9.6.4" data-path="data.html"><a href="data.html#stack-distance-maps-and-inspect-results"><i class="fa fa-check"></i><b>9.6.4</b> Stack distance maps and inspect results</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="data.html"><a href="data.html#spatial-statistics-at-segmento-level"><i class="fa fa-check"></i><b>9.7</b> Spatial statistics at <em>segmento</em> level</a><ul>
<li class="chapter" data-level="9.7.1" data-path="data.html"><a href="data.html#adjusting-the-coordinates-reference-system-for-raster-extaction-to-segmento"><i class="fa fa-check"></i><b>9.7.1</b> Adjusting the Coordinates Reference System for raster extaction to <em>segmento</em></a></li>
<li class="chapter" data-level="9.7.2" data-path="data.html"><a href="data.html#rasters"><i class="fa fa-check"></i><b>9.7.2</b> Rasters</a></li>
<li class="chapter" data-level="9.7.3" data-path="data.html"><a href="data.html#vectors"><i class="fa fa-check"></i><b>9.7.3</b> Vectors</a></li>
<li class="chapter" data-level="9.7.4" data-path="data.html"><a href="data.html#population-density"><i class="fa fa-check"></i><b>9.7.4</b> Population density</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="data.html"><a href="data.html#merging-the-covariates-with-the-ehpm-data"><i class="fa fa-check"></i><b>9.8</b> Merging the covariates with the EHPM data</a></li>
<li class="chapter" data-level="9.9" data-path="data.html"><a href="data.html#summary"><i class="fa fa-check"></i><b>9.9</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="http://flowminder.org/" target="blank">Released by Flowminder Foundation</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">High-resolution spatial modelling of population welfare in El Salvador, a coding tutorial</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="theory" class="section level1">
<h1><span class="header-section-number">Section 8</span> Appendix 1: Primer on INLA</h1>
<p>This section provides an introduction to high resolution mapping based on Bayesian geostatistical techniques. The aim is to provides an intuitive understanding of the models rather than an in-depth derivation and explanation of the theory underpinning them. This section is largely based on the books by Zuur et al. <span class="citation">(Zuur, Ieno, and Saveliev <a href="#ref-zuur2017beginner">2017</a>)</span> and Blangiardo and Cameletti <span class="citation">(Blangiardo and Cameletti <a href="#ref-blangiardo2015spatial">2015</a>)</span>.</p>
<div id="area-data-vs-point-referenced-data" class="section level2">
<h2><span class="header-section-number">8.1</span> Area data vs point-referenced data</h2>
<p>There are three types of spatial data <span class="citation">(Blangiardo and Cameletti <a href="#ref-blangiardo2015spatial">2015</a>)</span>:</p>
<ul>
<li><em>Area or lattice data</em>: the outcome variable to be modelled is a “random aggregate value over an aeral unit”<span class="citation">(Blangiardo and Cameletti <a href="#ref-blangiardo2015spatial">2015</a>)</span>. The difference between area and lattice data is that the latter is aggregated over a regular grid while the former is aggregated over a set of irregular polygons such as administrative boudaries. For instance, income aggregated at the <em>segmento</em> level are <em>area</em> data.</li>
<li><em>Point-referenced (or geostatistical) data</em>: the outcome variable to be modelled is “a random outcome at a specific location” <span class="citation">(Blangiardo and Cameletti <a href="#ref-blangiardo2015spatial">2015</a>)</span>, such as households’ income measured at the household location georeferenced with latitude and longitude coordinates.</li>
<li><em>Spatial point patterns</em>: the outcome variable to be modelled represents the occurence or not of an event at a given location. While <em>point referenced data</em> are random outcomes at a <em>specific location</em>, the location of the occurence or not are themselve random in <em>spatial point patterns</em>. For instance, road traffic accidents could be modelled as a <em>spatial point patterns</em>.</li>
</ul>
<p>The EHPM data were collected at the household level. They were then aggregated at the <em>segmento</em> level, they are hence <em>area data</em>. Nevertheless, we will also explore a model assuming they are <em>point-referenced data</em>, using the centroids of the <em>segmentos</em> as the location.</p>
<div id="bayesian-approach" class="section level3">
<h3><span class="header-section-number">8.1.1</span> Bayesian approach</h3>
<p>In the frequentist approach, the interest lies in estimating in the probabilty of the data, say income, given the parameters, <span class="math inline">\(P(data | \beta)\)</span>, i.e. one assumes the parameters exist. In a Bayesian approach, the interest lies in estimating the probability of the parameters given the data, <span class="math inline">\(P(\beta | data)\)</span>.</p>
<p>The Bayes theorem states that: <span class="math display">\[
\begin{aligned}
P(A&amp;B)=P(A|B)P(B)
\end{aligned}
\]</span></p>
<p><span class="math display">\[
\begin{aligned}
P(A|B)=\frac{P(B|A)P(A)}{P(B)}
\end{aligned}
\]</span> Replacing <span class="math inline">\(A\)</span> and <span class="math inline">\(B\)</span> with the quantity of interest, this gives:</p>
<p><span class="math display">\[
\begin{aligned}
P(\beta|data)=\frac{P(data|\beta)P(\beta)}{P(data)}
\end{aligned}
\]</span> where <span class="math inline">\(P(\beta|data)\)</span> is the posterior distribution of the parameter <span class="math inline">\(\beta\)</span>, <span class="math inline">\(P(data|\beta)\)</span> is the likelihood function of the data (e.g. a <em>Gamma</em> distribution for income), <span class="math inline">\(P(\beta)\)</span> is the the prior distribution of the <span class="math inline">\(\beta\)</span> parameter.</p>
<p>This can be expressed as:</p>
<p><span class="math display">\[
\begin{aligned}
P(\beta|data)\propto P(data|\beta)P(\beta)
\end{aligned}
\]</span> which reads as the posterior distribution of <span class="math inline">\(\beta\)</span> is proportional to <span class="math inline">\(P(data|\beta)P(\beta)\)</span>.</p>
<p>There are various strategies for getting <span class="math inline">\(P(data|\beta)P(\beta)\)</span>: simulation with a Monte Carlo or Markov Chain Monte Carlo algorythm, or deterministic approach via Integrated Nested Laplace Approximation (INLA) implemented in the INLA package.</p>
</div>
</div>
<div id="model-formulation" class="section level2">
<h2><span class="header-section-number">8.2</span> Model formulation</h2>
<p>The aim is to model the relationship between the <em>outcome variable</em> <span class="math inline">\(y\)</span>, for instance <em>segmento</em> median income, and a set of covariates <span class="math inline">\(\mathbf{X}\)</span>, the bold notation denoting that <span class="math inline">\(\mathbf{X}\)</span> is matrix with each column corresponding to a covariate.</p>
<div id="a-simple-model-not-accounting-for-spatial-dependence" class="section level3">
<h3><span class="header-section-number">8.2.1</span> A simple model not accounting for spatial dependence</h3>
<p>Ignoring for now the spatial dimension, this could be modeled as:</p>
<p><span class="math display">\[
\begin{aligned}
\mathbf{y}=\beta_{0}+\mathbf{X}\mathbf{\beta}+\mathbf{\epsilon}
\end{aligned}
\]</span></p>
<p><span class="math display">\[
\begin{aligned}
\mathbf{\epsilon}\sim N(0,\mathbf\Omega)
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\mathbf{y}\)</span> is a income vector of size <span class="math inline">\(n\)</span> for the <span class="math inline">\(n\)</span> <em>segmentos</em>, with each element <span class="math inline">\(y_{i}\)</span> of <span class="math inline">\(i=1, ... ,n\)</span> is the median income level in <em>segmemto</em> <span class="math inline">\(i\)</span>, <span class="math inline">\(\mathbf{X}\)</span> is the matrix of covariates with <span class="math inline">\(k\)</span> columns for the <span class="math inline">\(k\)</span> covariates and <span class="math inline">\(n\)</span> rows for the <span class="math inline">\(n\)</span> <em>segmentos</em>, <span class="math inline">\(\mathbf{\beta}\)</span> is the vector of <span class="math inline">\(k\)</span> parameters indicating the effect of each covariate on <span class="math inline">\(y\)</span>, <span class="math inline">\(\mathbf{\epsilon}\)</span> is the vector of error term of size <span class="math inline">\(n\)</span>, distributed normally with mean <span class="math inline">\(0\)</span> and variance-covariance defined by the matrix <span class="math inline">\(\mathbf{\Omega}\)</span>:</p>
<p><span class="math display">\[
\begin{aligned}
\mathbf\Omega = \left[\begin{array}
                       {rrrr}
                       \sigma^2 &amp; 0 &amp; \dots &amp; 0 \\\
                       0 &amp; \sigma^2 &amp; \dots &amp; \vdots \\\
                       \vdots &amp;  &amp;\ddots &amp; 0 \\\
                       0 &amp; \dots &amp; 0 &amp; \sigma^2
                       \end{array}\right]=\sigma^2\mathbf{I}
\end{aligned}
\]</span></p>
<p>with each off-diagonal element of <span class="math inline">\(\mathbf\Omega\)</span> representing the covariance of <span class="math inline">\(\epsilon_{i}\)</span> and <span class="math inline">\(\epsilon_{j}\)</span>, here set to <span class="math inline">\(0\)</span>, and <span class="math inline">\(\sigma^2\)</span> is the variance, which can hence be more consisely written as <span class="math inline">\(\sigma^2\mathbf{I}\)</span>, where <span class="math inline">\(\mathbf{I}\)</span> is an identity matrix where the diagonal elements are <span class="math inline">\(1\)</span> and the off-diagonal elements are <span class="math inline">\(0\)</span>.</p>
<p>The strong assumption of this model is that the term <span class="math inline">\(\epsilon_{i}\)</span> is idependently and identically distributed, often abreviated as <em>iid</em> distributed. Let us unpack this expression.</p>
<p><em>Independence.</em> The independence assumption implies that <span class="math inline">\(\epsilon_{i}\)</span> and <span class="math inline">\(\epsilon_{j}\)</span> for any two pairs of <em>segmento</em> <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> will be independent, even if <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> are neighbouring <em>segmentos</em>. This is represented by the covariance between <span class="math inline">\(\epsilon_{i}\)</span> and <span class="math inline">\(\epsilon_{j}\)</span> for <span class="math inline">\(i\neq j\)</span> being set to zero in <span class="math inline">\(\Omega\)</span>.</p>
<p>We can expect that it does not hold in the case of average income per <em>segmento</em>. For instance, let us imagine that a large company is recruiting its worksforce in <em>segmentos</em> <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span> and that no data is avaible in <span class="math inline">\(\mathbf{X}\)</span> to take this into account. Hence, <span class="math inline">\(\epsilon_{i}\)</span> and <span class="math inline">\(\epsilon_{j}\)</span> will be highly correlated, violating the independence assumption.</p>
<p><em>Identical distribution.</em> The are no symbol <span class="math inline">\(i\)</span> in the normal distribution <span class="math inline">\(N(\mu,\sigma^2)\)</span>, i.e. <span class="math inline">\(\epsilon_{i}\)</span> of all <em>segmento</em> are assumed to have a mean <span class="math inline">\(\mu\)</span> and a variance <span class="math inline">\(\sigma^2\)</span>. However, both the mean and the variance might vary across locations. For instance, one can expected that in an areas where most of the workforce is employed in agriculture, weather shocks might implies larger deviation from model prediction than in areas where most of the workforce is employed in the service sector, leading to different variance. Similarly, the model might systematically over- or under-predict in some areas, leading to <span class="math inline">\(\epsilon_{i}\)</span> with <span class="math inline">\(\mu\)</span> higher or under zero.</p>
<p>The violation of theses assumptions lead to incorrect estimate of the precison of the <span class="math inline">\(\mathbf{\beta}\)</span> parameters.</p>
<p>Furthermore, modelling explicilty the spatial dependence increase the predictive performance of the model; not taking into account the location implies throwing away a imporant piece of information.</p>
</div>
<div id="a-spatial-model" class="section level3">
<h3><span class="header-section-number">8.2.2</span> A spatial model</h3>
<p>In order to model explicitly the spatial dependence, an additional term is added to equation:</p>
<p><span class="math display">\[
\begin{aligned}
\mathbf{y}=\beta_{0}+\mathbf{X}\mathbf{\beta}+\mathbf{\epsilon}+\mathbf{\upsilon}
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\mathbf{\epsilon}\)</span> is distributed according to equation presented in the previous sub-section and <span class="math inline">\(\mathbf{\upsilon}\)</span> is a spatially correlated random effect distributed according:</p>
<p><span class="math display">\[
\begin{aligned}
\mathbf{\upsilon}\sim N(0,\mathbf\Sigma)
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\Sigma\)</span> is non-diagonal matrix given by:</p>
<p><span class="math display">\[
\begin{aligned}
\mathbf\Sigma = \left[\begin{array}
                       {rrrr}
                       \sigma_{u}^2 &amp;\phi_{1,2}^2 &amp; \dots &amp; \phi_{1,n}^2  \\\
                       \vdots &amp; \sigma_{u}^2 &amp; \dots &amp; \vdots \\\
                       \vdots &amp;  &amp;\ddots &amp; \phi_{n-1,n-1}^2 \\\
                       \dots &amp; \dots &amp; \dots &amp; \sigma_{u}^2
                       \end{array}\right]
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\sigma_{u}^2\)</span> is the variance of the <span class="math inline">\(\upsilon\)</span> and the <span class="math inline">\(\phi_{i,j}=corr(\upsilon_{i},\upsilon_{j})\neq0\)</span> is the correlation between the spatial random effect at location <span class="math inline">\(i\)</span> and <span class="math inline">\(j\)</span>.</p>
<p>The challenge now is to estimate the <span class="math inline">\(\phi_{i,j}\)</span> parameters. In the present case, the matrix <span class="math inline">\(\mathbf\Sigma\)</span> has a dimension of <span class="math inline">\(1664*1664\)</span>, i.e 2’768’896 parameters to estimates.</p>
</div>
<div id="estimating-the-model-with-aeral-data" class="section level3">
<h3><span class="header-section-number">8.2.3</span> Estimating the model with aeral data</h3>
<p>When working with aearal data, quantiyfying proximity with an euclidean distance is not adequate. First, one would need to figure out which point of the area should be considered when measuring the distance. One option could be to take some measure of the centre of each <em>segmentos</em>, however larger <em>segmentos</em> would then appear more remote than smaller one. The prefered approach with aeral data is hence slightly different: proximity is defined in terms of which <em>segmentos</em> do share a border.</p>
<p>For sack of simplicity, let us assume for the moment that the link function is a simple identity function such that <span class="math inline">\(g(\mu(s_{i}))=\mu(s_{i})\)</span>. <span class="math inline">\(\mu(s_{i})\)</span> can hence be expressed as:</p>
<p><span class="math display">\[
\begin{aligned}
\mu(s_{i})=\eta_{i}=\mathbf{X(s_{i})}\mathbf{\beta}+\upsilon_{i}
\end{aligned}
\]</span></p>
<p>A conditional autoregressive model (CAR) correlation function is used to model the spatial random effect <span class="math inline">\(\upsilon_{i}\)</span> (when dealing with point reference data, a Matern correlation and its SDPE expression is used instead, see next sub-section).</p>
<p>The spatial dependency is assumed to be Markovian in nature, i.e. the spatial correlation can be summarized by the spatial correlation between direct neighbours.</p>
<p>The distribution of each <span class="math inline">\(\upsilon_{i}\)</span> conditional on all other <span class="math inline">\(\upsilon\)</span> is expressed as:</p>
<p><span class="math display">\[
\begin{aligned}
\upsilon_{i}|\upsilon_{-i} \sim N(\sum_{j\neq i}^{N}c_{i,j}\upsilon_{j}, \sigma_{i}^{2}) 
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\({-i}\)</span> means all except <span class="math inline">\(i\)</span>. In the simplest model, called the intrinsic CAR, <span class="math inline">\(c_{i,j}\)</span> is <span class="math inline">\(1\)</span> if <span class="math inline">\(j\)</span> is neigbour of <span class="math inline">\(i\)</span> and <span class="math inline">\(0\)</span> otherwise. The conditional mean of each <span class="math inline">\(\upsilon_{i}\)</span> is hence an average of its direct neighbour.</p>
<p>The joint distribution of all <span class="math inline">\(\upsilon_{i}\)</span> is also Gaussian and written as:</p>
<p><span class="math display">\[
\begin{aligned}
\mathbf{\upsilon} \sim N(0, (\mathbf{I-C})^{-1}\mathbf{M}) 
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\mathbf{I}\)</span> is a matrix of <span class="math inline">\(1\)</span> and <span class="math inline">\(0\)</span>, <span class="math inline">\(\mathbf{C}\)</span> contains the <span class="math inline">\(c_{i,j}\)</span> and <span class="math inline">\(\mathbf{M}\)</span> is the covariance matrix. Various CAR model specify various function for <span class="math inline">\(\mathbf{C}\)</span> and <span class="math inline">\(\mathbf{M}\)</span>.</p>
<p>The homogeneous CAR model set:</p>
<p><span class="math display">\[
\begin{aligned}
\mathbf{C}=\phi\mathbf{A} \space \space \space \space \space \mathbf{M}=\mathbf{I}\sigma^{2}_{CAR}
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\mathbf{A}\)</span> is <span class="math inline">\(1\)</span> if area are neighbour and <span class="math inline">\(0\)</span> otherwise and <span class="math inline">\(\phi\)</span> needs to be estimated: larger it is, more corellated are the random effects. The instric car model set <span class="math inline">\(\phi\)</span> to 1.</p>
<p>One of the issues of CAR models is to lead to overfitting, i.e. model perform well on the training set of the data but poorly when doing out of sample prediction on the test set for validation purposes.</p>
<p>A solution is to smooth the pattern of the spatial random effects. With a lower <span class="math inline">\(\sigma^{2}_{CAR}\)</span>, the <span class="math inline">\(\upsilon\)</span> will vary less from neighbour to neigbour, leading to less overfit. In order to push the <span class="math inline">\(\sigma^{2}_{CAR}\)</span> to be lower, one can work on the priors. The standard way of doing this is by using penalized complexity priors whereby one specify the probability that <span class="math inline">\(\sigma^{2}_{CAR}\)</span> will be larger than a given number. To further limit the risk of overfitting. The standard way is to decompose the spatial random effects into one part which is spatially correlated and one part which is pure noise. This is called the modified Besag-York-Mollie <span class="citation">(Simpson et al. <a href="#ref-simpson2017penalising">2017</a>)</span>, which we will be using in the next section.</p>
</div>
<div id="estimating-the-model-with-point-referenced-data" class="section level3">
<h3><span class="header-section-number">8.2.4</span> Estimating the model with point-referenced data</h3>
<p>Let us now assume that the <em>segmentos</em> median income are point referenced data.</p>
<p>We now reconsider <span class="math inline">\(\mathbf{y}\)</span> as a random variable distributed at <span class="math inline">\(n\)</span> locations <span class="math inline">\(s_{1}, ...,s_{n}\)</span>. The income data can be considered as a sample of <span class="math inline">\(y(s_{1}), ..., y(s_{n})\)</span> from the random process <span class="math inline">\(\mathbf{y(s)}\)</span>.</p>
<p>Assuming that the <span class="math inline">\(y(s_{i})\)</span> are normally distributed, we have:</p>
<p><span class="math display">\[
\begin{aligned}
y(s_{i})\sim N(\mu(s_{i}),\sigma^2)
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(\mu(s_{i})\)</span> is expressed as a function of a structured additive predictor <span class="math inline">\(\eta_{i}\)</span>, such that <span class="math inline">\(g(\mu(s_{i}))=\eta_{i}\)</span>. For sack of simplicity, let us assume for the moment that the link function is a simple identity function such that <span class="math inline">\(g(\mu(s_{i}))=\mu(s_{i})\)</span>. <span class="math inline">\(\mu(s_{i})\)</span> can hence be expressed as:</p>
<p><span class="math display">\[
\begin{aligned}
\mu(s_{i})=\eta_{i}=\mathbf{X(s_{i})}\mathbf{\beta}+\upsilon_{i}+\epsilon_{i}
\end{aligned}
\]</span></p>
<p>Assuming the that <span class="math inline">\(\upsilon_{i}\)</span> is normally distributed, it makes it a Gaussian Field:</p>
<p><span class="math display">\[
\begin{aligned}
\upsilon_{i} \sim GF(\mathbf{0},\mathbf{\Sigma})
\end{aligned}
\]</span></p>
<p>where <span class="math inline">\(GF\)</span> stands for Gaussian fields.</p>
<p>Assuming that the covariance is Markovian in nature, i.e. only the spatial correlation can be summarized by the spatial correlation between direct neighbours, then the <span class="math inline">\(\upsilon_{i}\)</span> is distributed acording to a Gaussian Markov Random Field (GMRF):</p>
<p><span class="math display">\[
\begin{aligned}
\upsilon_{i} \sim GRMF(\mathbf{0},\mathbf{\Sigma})
\end{aligned}
\]</span></p>
<p>As <span class="math inline">\(\mathbf{\Sigma}\)</span> can be large (a matrix with more than 2 millions element in case of the <em>segmentos</em>), a determininistic structure is imposed on it in order to speed up the estimation,</p>
<p>When working with point referenced data, the variance-covariance <span class="math inline">\(\mathbf{\Sigma}\)</span> is expressed in terms of the Matern correlation function :</p>
<p><span class="math display">\[
\begin{aligned}
\mathbf{\Sigma}=\sigma_{\upsilon}^2 cor_{Matern}(\upsilon(s_{i}),\upsilon(s_{j}))
\end{aligned}
\]</span></p>
<p>The advantage of the Matern correlation function is that only a few parameters need to be estimated.</p>
<p>In order to further simplifiy the computation, the Matern correaltion can be re-expressed with a Stochastic Partial Differencial Equation (SPDE). Once the SPDE equation solved, one can use the SPDE parameters to solve the Matern correlation parameters.</p>
<p>In order to solve the SPDE, a spatial mesh is created on the point reference data. For each node of the mesh, we get a value <span class="math inline">\(w_{k}\)</span> via the finite element approach. The <span class="math inline">\(w_{k}\)</span> also form a GMRF. Once we have the <span class="math inline">\(w_{k}\)</span>, we can calculate the <span class="math inline">\(\upsilon_{i}\)</span> as a weighted sum of <span class="math inline">\(w_{k}\)</span> time <span class="math inline">\(a_{i,k}\)</span>s, where <span class="math inline">\(a_{i,k}\)</span> are the distance of <span class="math inline">\(s{i}\)</span> to node <span class="math inline">\(k\)</span>. This allows us to obtain the posterior distribution of the <span class="math inline">\(\upsilon_{i}\)</span>s.</p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-zuur2017beginner">
<p>Zuur, Alain F, Elena N Ieno, and Anatoly A Saveliev. 2017. <em>Beginner’s Guide to Spatial, Temporal, and Spatial-Temporal Ecological Data Analysis with R-Inla</em>.</p>
</div>
<div id="ref-blangiardo2015spatial">
<p>Blangiardo, Marta, and Michela Cameletti. 2015. <em>Spatial and Spatio-Temporal Bayesian Models with R-Inla</em>. John Wiley &amp; Sons.</p>
</div>
<div id="ref-simpson2017penalising">
<p>Simpson, Daniel, Håvard Rue, Andrea Riebler, Thiago G Martins, Sigrunn H Sørbye, and others. 2017. “Penalising Model Component Complexity: A Principled, Practical Approach to Constructing Priors.” <em>Statistical Science</em> 32 (1). Institute of Mathematical Statistics: 1–28.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="discussion.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"download": ["SV_mapping.pdf", "SV_mapping.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
